tables/mdm/mdm.go

```diff
@@ -160,20 +167,103 @@ func getMDMProfileStatus() (profileStatus, error) {
 	}, nil
 }
 
-func getDEPStatus() (depStatus, error) {
-	cmd := exec.Command("/usr/bin/profiles", "show", "-type", "enrollment")
-	out, err := cmd.Output()
+// Either get the live DEP capability status, or return from the cache if needed.
+func getDEPStatus(status profileStatus) (depStatus, error) {
+	// if we are enrolled via dep, we are by definion dep capable
+	if status.DEPEnrolled {
+		return depStatus{DEPCapable: true}, nil
+	}
+	var depstatus depStatus
+	hasAlreadyChecked := hasCheckedCloudConfigInPast24Hours()
+	if !hasAlreadyChecked {
+		cmd := exec.Command("/usr/bin/profiles", "show", "-type", "enrollment")
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			if strings.Contains(string(out), "Request too soon") {
+				depCapable := getCachedDEPStatus()
+				depstatus.DEPCapable = depCapable
+				return depstatus, nil
+			}
+			return depstatus, nil
+		}
+
+		lines := bytes.Split(out, []byte("\n"))
+
+		if len(lines) > 3 {
```

Comment:
optional: Please add a comment explaining the significance of `len(lines) > 3` here.